memCache/entryCache.go

```diff
@@ -433,3 +215,216 @@ func (e *Entry) timeToUpdate() bool {
 	}
 	return false
 }
+
+type EntryCache struct {
+	log            *log.Logger
+	clk            clock.Clock
+	requestTimeout time.Duration
+	entries        map[string]*Entry   // one-to-one map keyed on name -> entry
+	lookupMap      map[[32]byte]*Entry // many-to-one map keyed on sha256 hashed OCSP requests -> entry
```

Comment:
Using just `entries` requires ranging over the full map and hashing + checking each issuer and serial to match against the request. Sacrificing some memory lets us do a direct lookup of the request -> entry.